odoo/addons/base/models/ir_autovacuum.py

```diff
@@ -37,10 +43,32 @@ def _gc_user_logs(self):
         _logger.info("GC'd %d user log entries", self._cr.rowcount)
 
     @api.model
-    def power_on(self, *args, **kwargs):
+    def run_garbage_collectors(self):
         if not self.env.is_admin():
             raise AccessDenied()
-        self.env['ir.attachment']._file_gc()
-        self._gc_transient_models()
-        self._gc_user_logs()
+
+        for model in self.env.values():
+            for method in model._autovacuum:
+                _logger.info("Executing %s.%s", model, method)
+                try:
+                    getattr(model, method)()
+                except Exception:
+                    _logger.warning("Could not vacuum model %s", model, exc_info=True)
+                else:
+                    self.env.cr.commit()
```

Comment:
Beware, as `cr.commit()` will flush stuff like recomputed fields.
I suggest to commit inside the `try-except`, and don't forget to rollback if it fails!
```suggestion
                try:
                    getattr(model, method)()
                    self.env.cr.commit()
                except Exception:
                    _logger.warning("Failed %s.%s", model, method, exc_info=True)
                    self.env.cr.rollback()
```